diff --git a/package/boot/uboot-tools/uboot-envtools/files/airoha_an7581 b/package/boot/uboot-tools/uboot-envtools/files/airoha_an7581
index c0a5c9fb4..fb850a276 100644
--- a/package/boot/uboot-tools/uboot-envtools/files/airoha_an7581
+++ b/package/boot/uboot-tools/uboot-envtools/files/airoha_an7581
@@ -13,6 +13,7 @@ board=$(board_name)
 
 case "$board" in
 gemtek,w1700k-ubi|\
+gemtek,xr1710g-ubi|\
 nokia,xg-040g-md-ubi)
 	ubootenv_add_ubi_default
 	;;
