diff --git a/package/boot/uboot-tools/uboot-envtools/files/airoha_an7581 b/package/boot/uboot-tools/uboot-envtools/files/airoha_an7581
index 29e6c398868a41..b890042840170b 100644
--- a/package/boot/uboot-tools/uboot-envtools/files/airoha_an7581
+++ b/package/boot/uboot-tools/uboot-envtools/files/airoha_an7581
@@ -12,7 +12,8 @@ touch /etc/config/ubootenv
 board=$(board_name)
 
 case "$board" in
-gemtek,w1700k-ubi)
+gemtek,w1700k-ubi|\
+gemtek,xr1710g-ubi)
 	ubootenv_add_ubi_default
 	;;
 esac
