diff --git a/target/linux/airoha/an7581/base-files/etc/init.d/airoha_fan b/target/linux/airoha/an7581/base-files/etc/init.d/airoha_fan
index ab9eeca83ae695..6ec385ad7f4fd6 100755
--- a/target/linux/airoha/an7581/base-files/etc/init.d/airoha_fan
+++ b/target/linux/airoha/an7581/base-files/etc/init.d/airoha_fan
@@ -21,7 +21,8 @@ find_nct7802()
 boot()
 {
 	case $(board_name) in
-	gemtek,w1700k-ubi)
+	gemtek,w1700k-ubi|\
+	gemtek,xr1710g-ubi)
 		hwmon=$(find_nct7802)
 
 		# Temporarily disable automatic fan control to configure curve
