diff --git a/package/boot/uboot-airoha/Makefile b/package/boot/uboot-airoha/Makefile
index 52b95c68edc70e..cc8cfe2c8ab73e 100644
--- a/package/boot/uboot-airoha/Makefile
+++ b/package/boot/uboot-airoha/Makefile
@@ -54,6 +54,14 @@ define U-Boot/an7581_gemtek_w1700k
   UBOOT_IMAGE:=u-boot.bin
 endef
 
+define U-Boot/an7581_gemtek_xr1710g
+  NAME:=Gemtek XR1710G (UBI)
+  UBOOT_CONFIG:=an7581_xr1710g
+  BUILD_DEVICES:=gemtek_xr1710g-ubi
+  BUILD_SUBTARGET:=an7581
+  UBOOT_IMAGE:=u-boot.bin
+endef
+
 define U-Boot/an7581_nokia_valyrian
   NAME:=Nokia Valyrian
   UBOOT_CONFIG:=an7581_nokia_valyrian
@@ -69,6 +77,7 @@ UBOOT_TARGETS := \
 	an7581_rfb \
 	an7583_rfb \
 	an7581_gemtek_w1700k \
+	an7581_gemtek_xr1710g \
 	an7581_nokia_valyrian
 
 UBOOT_CUSTOMIZE_CONFIG := \
