diff --git a/package/boot/uboot-airoha/Makefile b/package/boot/uboot-airoha/Makefile
index 5db35c03e..2da4651e9 100644
--- a/package/boot/uboot-airoha/Makefile
+++ b/package/boot/uboot-airoha/Makefile
@@ -24,6 +24,13 @@ define U-Boot/an7581_gemtek_w1700k
   BUILD_SUBTARGET:=an7581
 endef
 
+define U-Boot/an7581_gemtek_xr1710g
+  NAME:=Gemtek XR1710G (UBI)
+  UBOOT_CONFIG:=an7581_xr1710g
+  BUILD_DEVICES:=gemtek_xr1710g-ubi
+  BUILD_SUBTARGET:=an7581
+endef
+
 define U-Boot/an7581_nokia_valyrian
   NAME:=Nokia Valyrian
   UBOOT_CONFIG:=an7581_nokia_valyrian
@@ -83,6 +90,7 @@ endef
 
 UBOOT_TARGETS := \
 	an7581_gemtek_w1700k \
+	an7581_gemtek_xr1710g \
 	an7581_nokia_valyrian \
 	an7581_nokia_xg-040g-md \
 	an7581_rfb \
